### DISCUSSION TITLE
High CPU usage of "Capture process" on h265 camera. Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by LaurenceGough** [blakeblackshear/frigate#5926] [Ask A Question]
Hi all,

I am noticing quite high CPU usage on "Capture process" of one of my cameras, my only h265 one. 

ffmpeg 32.6% 
Capture 27.6%
Detect 6.0% (just included for info, I am using a USB Coral).

Whereas on my H264 camera it is 10% ffmpeg and 2.7% capture.

The lower usage one is 5fps, vs 8fps for the higher use one, and the resolution is higher at 1080p for detect for the higher use camera.

If I turn off hardware acceleration (hwaccel -drm) then CPU usage of ffmpeg process goes to around 85%+ for the h265 camera.

I understand ffmpeg would be higher for this codec but I was just wondering if anything could be done to lower the capture usage?

The resolution is matching the input RTSP stream (verified in VLC too).

Is it due to a limitation of the h265 / raspberry pi 4 HW decoding or something else? 

I use h265 mainly to save massively on the storage requirements.. well allowing longer retention.

I believe Frigate uses one thread per a camera (maybe more), so I guess this means this is using around half a core just at "idle".

I know the Pi 4 is not the most powerful device in the world, I just wanted to learn a bit more about this process and if it's expected.

Thanks

**Comment by blakeblackshear** [comment]
What do you see for the camera_fps on the system page? Does it match?

**Reply by LaurenceGough** [reply]
It says 8.1 which is odd as the camera and vlc say 8. 

I set the FPS in the config from 8 to 8.1 and restarted but the CPU usage is around the same. 

I'll try another FPS setting on the camera and get back to you. 

Thanks

**Reply by NickM-27** [reply]
It's just an average, not a direct calculation. Definitely leave it as whole numbers.
